backend/azure/dtmb/internal/utils/connectworker.go

```diff
@@ -0,0 +1,248 @@
+package utils
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/microsoft/durabletask-go/backend/azure/dtmb/internal/backend/v1"
+	"google.golang.org/grpc/metadata"
+)
+
+var UserAgent = "dev/1"
+
+func ConnectWorker(
+	ctx context.Context,
+	testID string,
+	taskHub string,
+	worker backend.TaskHubWorkerClient,
+	orchestratorFns OrchestratorFnList,
+	activityFns ActivityFnList,
+	serverMsgChan chan<- *backend.ConnectWorkerServerMessage,
+	clientMsgChan <-chan *backend.ConnectWorkerClientMessage,
+	readyCb func(),
+	debug bool,
+) error {
+	defer close(serverMsgChan)
+
+	// Establish the ConnectWorker stream
+	streamCtx := metadata.AppendToOutgoingContext(ctx,
+		"taskhub", taskHub,
+	)
+	stream, err := worker.ConnectWorker(streamCtx)
+	if err != nil {
+		return fmt.Errorf("error starting ConnectWorker: %w", err)
+	}
+
+	// Send the initial message
+	err = stream.Send(&backend.ConnectWorkerClientMessage{
+		Message: &backend.ConnectWorkerClientMessage_EstablishWorkerConnection{
+			EstablishWorkerConnection: &backend.EstablishWorkerConnectionMessage{
+				Version:              UserAgent,
+				ActivityFunction:     activityFns.ToProto(),
+				OrchestratorFunction: orchestratorFns.ToProto(),
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("error sending EstablishWorkerConnectionMessage message: %w", err)
+	}
+
+	// Wait for the first message
+	timeout := time.NewTimer(5 * time.Second)
+	configReceived := make(chan error, 1)
+	var wc *backend.WorkerConfiguration
+	go func() {
+		msg, err := stream.Recv()
+		if err != nil {
+			configReceived <- err
+			return
+		}
+
+		wc = msg.GetWorkerConfiguration()
+		if wc == nil {
+			configReceived <- errors.New("received unexpected message")
+			return
+		}
+
+		log.Printf("[%s] Received configuration message: %v", testID, wc)
```

Comment:
Remove logs

Reply:
Later! Right now I wanted just a high level review from you or what the specific calls to/from the backend